google-api-client/src/main/java/com/google/api/client/googleapis/javanet/GoogleNetHttpTransport.java

```diff
@@ -51,10 +56,44 @@ static HttpTransport newProxyTransport() throws GeneralSecurityException, IOExce
    */
   public static NetHttpTransport newTrustedTransport()
       throws GeneralSecurityException, IOException {
-    return new NetHttpTransport.Builder().trustCertificates(GoogleUtils.getCertificateTrustStore())
-        .build();
+    return newTrustedTransport(null, "");
   }
 
-  private GoogleNetHttpTransport() {
+  /**
+   * {@link Beta} <br>
+   * Returns a new instance of {@link NetHttpTransport} that uses
+   * {@link GoogleUtils#getCertificateTrustStore()} for the trusted certificates using
+   * {@link com.google.api.client.http.javanet.NetHttpTransport.Builder#trustCertificates(KeyStore)}.
+   * If `GOOGLE_API_USE_CLIENT_CERTIFICATE` environment variable is set to "true",
+   * the function uses the provided mtlsKeyStore or the default key store from
+   * {@link Utils#loadDefaultMtlsKeyStore()} to create the transport. If either key
+   * store exists, then the created transport is mutual TLS. The provided key store
+   * takes precedence over the default one. 
+   * 
+   * @param mtlsKeyStore KeyStore for mutual TLS client certificate and private key
+   * @param mtlsKeyStorePassword KeyStore password
+   * @since 1.31   
+   */
+  @Beta
+  public static NetHttpTransport newTrustedTransport(KeyStore mtlsKeyStore, String mtlsKeyStorePassword)
```

Comment:
This override is not intended to be general purpose. We need this override to control whether the keystore should be used for mtls, even if it is explicitly provided by the user. Python/Go has a similar override for the same functionality. We will probably need this control for a while until all mtls endpoints are GA. For googleapis, users should use this override in case they want their own certificate (even though this is discouraged).

Reply:
If this is necessary, with the refactor to an `MtlsProvider`, we can provide a custom provider where the user can provide their own `KeyStore` and make the `newTrustedTransport(MtlsProvider)` public.

Reply:
Now I think it, it should be fine just using the default cert.